flows/vibe_check.py

```diff
@@ -0,0 +1,537 @@
+        assert isinstance(selected_passages, pd.DataFrame)
+        for idx, (_, row) in enumerate(selected_passages.iterrows()):
+            text = str(row["text_block.text"])
+            predicted_spans = classifier.predict(text)
+            text_block_metadata = {str(k): str(v) for k, v in row.to_dict().items()}
+            labelled_passage = LabelledPassage(
+                text=text,
+                spans=predicted_spans,
+                metadata=text_block_metadata,
+            )
+            labelled_passages.append(labelled_passage)
+
+            # Update progress every 50 passages (or on the last passage)
+            passage_num = idx + 1
+            if passage_num % 50 == 0 or passage_num == n_passages:
+                progress = (passage_num / n_passages) * 100
+                update_progress_artifact(
+                    progress_id,  # type: ignore
+                    progress=progress,
+                    description=f"Processed passage {passage_num}/{n_passages}",
+                )
+
+        logger.info(f"Generated {len(labelled_passages)} labelled passages")
+
+        # before uploading the passages, we should shuffle them
+        random.shuffle(labelled_passages)
+
+        bucket_name = _get_bucket_name_from_ssm()
+        output_prefix = Path(wikibase_id) / classifier.id
+        logger.info(f"Outputs will be stored in s3://{bucket_name}/{output_prefix}")
+
+        # Push results for this concept to S3
+        jsonl_string = "\n".join(
+            [
+                json.dumps(
+                    {
+                        "marked_up_text": labelled_passage.get_highlighted_text(
+                            start_pattern='<span class="prediction-highlight">',
+                            end_pattern="</span>",
+                        ),
+                        **json.loads(labelled_passage.model_dump_json()),
+                    }
+                )
+                for labelled_passage in labelled_passages
+            ]
+        )
+
+        logger.info(f"Pushing predictions to S3: {output_prefix / 'predictions.jsonl'}")
+        push_object_bytes_to_s3(
+            s3_client=s3_client,
+            key=output_prefix / "predictions.jsonl",
+            data=jsonl_string.encode("utf-8"),
+        )
+
+        logger.info(f"Pushing concept data to S3: {output_prefix / 'concept.json'}")
+        push_object_bytes_to_s3(
+            s3_client=s3_client,
+            key=output_prefix / "concept.json",
+            data=concept.model_dump_json().encode("utf-8"),
+        )
+
+        logger.info(
+            f"Pushing classifier metadata to S3: {output_prefix / 'classifier.json'}"
+        )
+        push_object_bytes_to_s3(
+            s3_client=s3_client,
+            key=output_prefix / "classifier.json",
+            data=json.dumps(classifier_metadata).encode("utf-8"),
+        )
+
+        n_positive_passages = sum(1 for passage in labelled_passages if passage.spans)
```

Comment:
meta question i've been considering: how do we denote a false, passage-level prediction or annotation as opposed to a lack of prediction? in one version of my active learning code I decided that was a `Span` with `prediction_probability=0` but that felt wrong. if it was right, it'd break this. (this was my bug with active learning that lost me a day two weeks ago!)